core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -73,17 +73,17 @@ static ClientRequestContext current() {
      * Returns the client-side context of the {@link Request} that is being handled in the current thread.
      *
      * @return the {@link ClientRequestContext} available in the current thread, or {@code null} if unavailable.
-     * @throws IllegalStateException if the current context is not a {@link ClientRequestContext}.
      */
     @Nullable
     static ClientRequestContext currentOrNull() {
         final RequestContext ctx = RequestContext.currentOrNull();
         if (ctx == null) {
             return null;
         }
-        checkState(ctx instanceof ClientRequestContext,
-                   "The current context is not a client-side context: %s", ctx);
-        return (ClientRequestContext) ctx;
+        if (ctx instanceof ClientRequestContext) {
+            return (ClientRequestContext) ctx;
+        }
+        return null;
```

Comment:
Can be simplified into:

```
if (ctx instanceof ClientRequestContext) {
    return (...) ctx;
} else {
    return null;
}
```

i.e. no need to check if `null`.

Reply:
Thanks fixed. 😉